### PR TITLE
Make HRTF panning k-rate

### PR DESCRIPTION
**Pull request by padenot**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -8739,23 +8743,22 @@ <h4 id="Spatialzation-equal-power-panning">
             <a><code>PannerNode</code></a> when the <a href=
             "#widl-PannerNode-panningModel"><code>panningModel</code></a>
             attribute is set to <code>"equalpower"</code>, in which case the
-            the <em>elevation</em> value is ignored.
+            the <em>elevation</em> value is ignored. This algorithm MUST be
+            implemented using <a>k-rate</a> parameters.
```
Comment (rtoy) [inline index.html]: Don't you mean a-rate here?

File: index.html [hunk 2]
```diff
@@ -8739,23 +8743,22 @@ <h4 id="Spatialzation-equal-power-panning">
...
                 "#azimuth-elevation">azimuth and elevation</a> section.
               </p>
-            </li>
-            <li>
+              </li>
+              <li>
               <p>
-                The <em>azimuth</em> value is first contained to be within the
-                range [-90, 90] according to:
+              The <em>azimuth</em> value is first contained to be within the
```
Comment (rtoy) [inline index.html]: Indentation was changed here.  Doesn't tidy normally indent p tags by two?

File: index.html [hunk 3]
```diff
@@ -8790,91 +8793,96 @@ <h4 id="Spatialzation-equal-power-panning">
...
-if (pan &lt;= 0)
-  x = pan + 1;
-else
-  x = pan;
-        
-</pre>
-            </li>
-          </ol>
-          <p>
+    if (pan &lt;= 0)
```
Comment (rtoy) [inline index.html]: Why indent? Not really part of this change is it?

File: index.html [hunk 4]
```diff
@@ -5202,24 +5219,32 @@ <h2 id="AudioWorkerParamDescriptor">
             source is pointing in 3D Cartesian coordinate space. Depending on
             how directional the sound is (controlled by the <b>cone</b>
             attributes), a sound pointing away from the listener can be very
-            quiet or completely silent. The default value is 1. This parameter
-            is a-rate.
+            quiet or completely silent. The default value is 1. The default
+            value is 0. This parameter is <a>a-rate</a> when <a
```
Comment (rtoy) [inline index.html]: Too much cut-n-paste; remove the part about default is 0 because the default is 1.

File: index.html [hunk 5]
```diff
@@ -5202,24 +5219,32 @@ <h2 id="AudioWorkerParamDescriptor">
...
+            <code>"equalpower"</code>, <a>k-rate</a> otherwise.
           </dd>
           <dt>
             readonly attribute AudioParam orientationY
           </dt>
           <dd>
             Describes the y component of the vector of the direction the audio
             source is pointing in 3D cartesian coordinate space. The default
-            value is 0. This parameter is a-rate.
+            value is 0. The default value is 0. This parameter is
```
Comment (rtoy) [inline index.html]: Remove redundant default is 0 sentence.

File: index.html [hunk 6]
```diff
@@ -5202,24 +5219,32 @@ <h2 id="AudioWorkerParamDescriptor">
...
+            otherwise.
           </dd>
           <dt>
             readonly attribute AudioParam orientationZ
           </dt>
           <dd>
             Describes the Z component of the vector of the direction the audio
             source is pointing in 3D cartesian coordinate space. The default
-            value is 0. This parameter is a-rate.
+            value is 0. The default value is 0. This parameter is
```
Comment (rtoy) [inline index.html]: Remove redundant default is 0 sentence.

**Comment by joeberkovitz** [comment]
@padenot I think we are waiting on your response to @rtoy's suggested edits.

**Comment by padenot** [comment]
Those commits address the comments and are rebased against master.

File: index.html [hunk 7]
```diff
@@ -6242,6 +6242,10 @@ <h2 id="AudioWorklet-Examples">
           <dd>
             A simple and efficient spatialization algorithm using equal-power
             panning.
+            <div class="note">
+              When this panning model is used, all the <a>AudioParam</a> used
```
Comment (rtoy) [inline index.html]: Typo: "AudioParam" -> "AudioParams"

File: index.html [hunk 8]
```diff
@@ -6250,6 +6254,10 @@ <h2 id="AudioWorklet-Examples">
             A higher quality spatialization algorithm using a convolution with
             measured impulse responses from human subjects. This panning method
             renders stereo output.
+            <div class="note">
+              When this panning model is used, all the <a>AudioParam</a> used
```
Comment (rtoy) [inline index.html]: AudioParam->AudioParams

File: index.html [hunk 9]
```diff
@@ -6688,8 +6708,12 @@ <h2 id="AudioListener">
           <dd>
             <p>
               Sets the x coordinate position of the audio listener in a 3D
-              Cartesian coordinate space. The default value is 0. This
-              parameter is a-rate.
+              Cartesian coordinate space. <a><code>PannerNode</code></a>
+              objects use this position relative to individual audio sources
```
Comment (rtoy) [inline index.html]: I don't think this should be here since we're only talking about the x coordinate. This should be moved somewhere else, like what was done for forwardXYZ for the listener.

File: index.html [hunk 10]
```diff
@@ -6709,17 +6736,37 @@ <h2 id="AudioListener">
...
+              used to determine the orientation of the listener. In simple
+              human terms, the <code>forward</code> vector represents which
+              direction the person's nose is pointing. The <code>up</code>
+              vector represents the direction the top of a person's head is
+              pointing. These values are expected to be linearly independent
+              (at right angles to each other), and unpredictable behavior may
+              result if they are not. For normative requirements of how these
+              values are to be interpreted, see the <a href=
+              "#Spatialization">spatialization section</a>.
+            </p>
```
Comment (rtoy) [inline index.html]: This paragraph is already at https://webaudio.github.io/web-audio-api/#AudioListener, 3rd paragraph, so this should be removed.

File: index.html [hunk 11]
```diff
@@ -10558,25 +10615,24 @@ <h4 id="Spatialzation-equal-power-panning">
             <a><code>PannerNode</code></a> when the <a href=
             "#widl-PannerNode-panningModel"><code>panningModel</code></a>
             attribute is set to <code>"equalpower"</code>, in which case the
-            the <em>elevation</em> value is ignored.
-          </p>
-          <p>
-            For a <a><code>PannerNode</code></a>, the following algorithm MUST
-            be implemented.
+            the <em>elevation</em> value is ignored. This algorithm MUST be
+            implemented using <a>k-rate</a> parameters.
```
Comment (rtoy) [inline index.html]: a-rate here, right since it's equalpower?

File: index.html [hunk 12]
```diff
@@ -10586,114 +10642,120 @@ <h4 id="Spatialzation-equal-power-panning">
...
   } else { // 0 ~ 90
     // Transform the azimuth value from [0, 90] degrees into the range [-90, 90].
     x = azimuth / 90;
   }
-        
-</pre>
+
+              </pre>
+                </li>
+              </ol>
```
Comment (rtoy) [inline index.html]: This indentation seems wrong here.  Did tidy do the right thing here?

File: index.html [hunk 13]
```diff
@@ -10586,114 +10642,120 @@ <h4 id="Spatialzation-equal-power-panning">
...
+                  </p>
+                  <pre>
+    if (pan &lt;= 0)
+      x = pan + 1;
+    else
+      x = pan;
+
+              </pre>
+                </li>
+              </ol>
```
Comment (rtoy) [inline index.html]: What's really changing here? And why is this part of the change for the a-rate spec for the PannerNode?

Comment (padenot) [inline index.html]: We need to spell out the algorithm for each sample instead of for each block, so this change the indentation level so this looks like a big change.

If I go a `git diff -w HEAD^^^` in my terminal, that is, diff against master without whitespace changes, it looks better, but I don't think github supports that.

**Comment by rtoy** [comment]
lgtm.
